pkg/plugin/processor/builtin/unwrap/unwrap_opencdc.go

```diff
@@ -0,0 +1,293 @@
+	sdk.UnimplementedProcessor
+
+	logger      log.CtxLogger
+	fieldRefRes sdk.ReferenceResolver
+}
+
+func NewOpenCDCProcessor(logger log.CtxLogger) sdk.Processor {
+	return &openCDCProcessor{logger: logger}
+}
+
+func (u *openCDCProcessor) Specification() (sdk.Specification, error) {
+	return sdk.Specification{
+		Name:    "unwrap.opencdc",
+		Summary: "A processor that unwraps the OpenCDC record saved in one of record's fields.",
+		Description: `
+The unwrap.opencdc processors unwraps the OpenCDC record saved in one of the record's fields.
+
+This is useful in situations where a record goes through intermediate systems before 
+being written to a final destination. In these cases, the original OpenCDC record is
+part of the payload read from the intermediate system and needs to be unwrapped before
+being written to the destination.
+'`,
+		Version:    "v0.1.0",
+		Author:     "Meroxa, Inc.",
+		Parameters: openCDCConfig{}.Parameters(),
+	}, nil
+}
+
+func (u *openCDCProcessor) Configure(ctx context.Context, m map[string]string) error {
+	cfg := openCDCConfig{}
+	err := sdk.ParseConfig(ctx, m, &cfg, cfg.Parameters())
+	if err != nil {
+		return cerrors.Errorf("failed parsing configuration: %w", err)
+	}
+
+	rr, err := sdk.NewReferenceResolver(cfg.Field)
+	if err != nil {
+		return cerrors.Errorf("invalid reference: %w", err)
+	}
+
+	u.fieldRefRes = rr
+	return nil
+}
+
+func (u *openCDCProcessor) Open(context.Context) error {
+	return nil
+}
+
+func (u *openCDCProcessor) Process(_ context.Context, records []opencdc.Record) []sdk.ProcessedRecord {
+	out := make([]sdk.ProcessedRecord, 0, len(records))
+	for _, rec := range records {
+		proc := u.processRecord(rec)
+		out = append(out, proc)
+		if _, ok := proc.(sdk.ErrorRecord); ok {
+			return out
+		}
+	}
+
+	return out
+}
+
+func (u *openCDCProcessor) processRecord(rec opencdc.Record) sdk.ProcessedRecord {
+	ref, err := u.fieldRefRes.Resolve(&rec)
+	if err != nil {
+		return sdk.ErrorRecord{Error: cerrors.Errorf("failed resolving record reference: %w", err)}
+	}
+
+	var data opencdc.StructuredData
+	switch v := ref.Get().(type) {
+	case opencdc.RawData:
+		// unmarshal raw data to structured
+		if err := json.Unmarshal(v.Bytes(), &data); err != nil {
+			return sdk.ErrorRecord{Error: cerrors.Errorf("failed to unmarshal raw data as JSON: %w", err)}
+		}
+	case opencdc.StructuredData:
+		data = v
+	case nil:
+		return sdk.ErrorRecord{Error: cerrors.New("field to unmarshal is nil")}
+	default:
+		return sdk.ErrorRecord{Error: cerrors.Errorf("unexpected data type %T", v)}
+	}
+
+	opencdcRec, err := u.unmarshalRecord(data)
+	if err != nil {
+		return sdk.ErrorRecord{Error: cerrors.Errorf("failed unmarshalling record: %w", err)}
+	}
+	opencdcRec.Position = rec.Position
+
+	// Position is the only key we preserve from the original record to maintain the reference respect other messages
+	// that will be coming from in the event of chaining pipelines (e.g.: source -> kafka, kafka -> destination)
```

Comment:
Nitpick - I'd expect these comments above the line where we update the position.